tests/pytorch/test_stream.py

```diff
@@ -34,6 +62,64 @@ def test_basics():
     gg = g.to(device=F.ctx())
     OPS.copy_u_sum(gg, xx)
 
+@unittest.skipIf(F._default_context_str == 'cpu', reason="stream only runs on GPU.")
+def test_set_get_stream():
+    current_stream = torch.cuda.current_stream()
+    # test setting another stream
+    s = torch.cuda.Stream(device=F.ctx())
+    torch.cuda.set_stream(s)
+    assert to_dgl_stream_handle(s).value == _dgl_get_stream(to_dgl_context(F.ctx())).value
+    # revert to default stream
+    torch.cuda.set_stream(current_stream)
+
+@unittest.skipIf(F._default_context_str == 'cpu', reason="stream only runs on GPU.")
+def test_record_stream_smoke():
+    # smoke test to ensure it can work
+    stream = torch.cuda.Stream()
+    with torch.cuda.stream(stream):
+        t = nd.array(np.array([1., 2., 3., 4.]), ctx=nd.gpu(0))
+        g = rand_graph(10, 20, device=F.cuda())
+    # NDArray is an internal object, just use DGLStreamHandle
+    t.record_stream(to_dgl_stream_handle(torch.cuda.current_stream()))
+    g.record_stream(torch.cuda.current_stream())
+
+@unittest.skipIf(F._default_context_str == 'cpu', reason="stream only runs on GPU.")
+# borrowed from PyTorch, test/test_cuda.py: test_record_stream()
+def test_record_stream():
+    cycles_per_ms = _get_cycles_per_ms()
+
+    t = nd.array(np.array([1., 2., 3., 4.], dtype=np.float32), ctx=nd.cpu())
+    t.pin_memory_()
+    result = nd.empty([4], ctx=nd.gpu(0))
+    stream = torch.cuda.Stream()
+    ptr = [None]
+
+    # Performs the CPU->GPU copy in a background stream
+    def perform_copy():
+        with torch.cuda.stream(stream):
+            tmp = t.copyto(nd.gpu(0))
+            ptr[0] = F.from_dgl_nd(tmp).data_ptr()
+        torch.cuda.current_stream().wait_stream(stream)
```

Comment:
What's the difference between `torch.cuda.current_stream()` and `stream` which is `torch.cuda.Stream()` here? Can they be same?

Reply:
`stream = torch.cuda.Stream()` will create a new stream. So code in
```
with torch.cuda.stream(stream):
    ...
```
will run in a different stream while code outside this with context still runs in the `torch.cuda.current_stream()`.